### PR TITLE
Sema: Lift flag requirement for access levels on imports now that SE-0409 has been accepted

### PR DESCRIPTION
**Pull request by xymus**
The proposal [SE-0409 Access level on imports](https://github.com/apple/swift-evolution/blob/main/proposals/0409-access-level-on-imports.md) has been accepted, let's lift the flag requirement!

Also adds the upcoming feature flag `InternalImportsByDefault` for imports to default to internal. This is the Swift 6 behavior that can be enabled in Swift 5 using this new flag. The logic is cleaned but by moving all the Swift 6 specific logic being this flag.

rdar://107566870

### CODE REVIEW

**Comment by xymus** [comment]
@swift-ci Please smoke test

File: test/ModuleInterface/access-level-import-swiftinterfaces.swift [hunk 1]
```diff
@@ -21,30 +21,57 @@
...
 
-// RUN: %FileCheck %s < %t/MultiFiles.swiftinterface
-// RUN: %FileCheck %s < %t/MultiFiles.private.swiftinterface
+// RUN: %FileCheck --check-prefixes=CHECK,CHECK-5 %s < %t/MultiFiles.swiftinterface
+// RUN: %FileCheck --check-prefixes=CHECK,CHECK-5 %s < %t/MultiFiles.private.swiftinterface
+
+/// Swift 6 mode.
+// RUN: %target-swift-frontend -typecheck %t/Client.swift -I %t \
+// RUN:   -package-name TestPackage -module-name Client_Swift6 \
+// RUN:   -enable-library-evolution -swift-version 6 \
```
Comment (tshortli) [inline test/ModuleInterface/access-level-import-swiftinterfaces.swift:50]: Don't forget `REQUIRES: asserts`.

File: include/swift/Basic/Features.def [hunk 2]
```diff
@@ -117,6 +117,7 @@ UPCOMING_FEATURE(BareSlashRegexLiterals, 354, 6)
 UPCOMING_FEATURE(ExistentialAny, 335, 6)
 UPCOMING_FEATURE(ImportObjcForwardDeclarations, 384, 6)
 UPCOMING_FEATURE(DisableOutwardActorInference, 401, 6)
+UPCOMING_FEATURE(InternalImports, 409, 6)
```
Comment (tshortli) [inline include/swift/Basic/Features.def:120]: This feature name makes it sound like you need to specify it in order to write `internal import`. What about something like `InternalImportsByDefault`?

**Review by tshortli — APPROVED** [review]
Exciting!

**Comment by tshortli** [comment]
This is just a nice to have, but it could be helpful to emit a warning for projects that already adopted `-enable-experimental-feature AccessLevelOnImports` to let them know they can remove the flag.

File: test/ModuleInterface/access-level-import-swiftinterfaces.swift (hunk 1 above)
Comment (xymus) [reply test/ModuleInterface/access-level-import-swiftinterfaces.swift:50]: Good catch, thanks!

**Comment by xymus** [comment]
I think we'll get the warning automatically when we delete the flag. I'm holding off on that because the flag still has one effect, promoting to errors imports of a non-resilient module from a resilient module. It's a small thing but might as well keep it around until most projects move to Swift 6 mode.

**Review by elsh — APPROVED** [review]
🎆

File: include/swift/Basic/Features.def (hunk 2 above)
Comment (xymus) [reply include/swift/Basic/Features.def]: Good point, I've updated the PR to use `InternalImportsByDefault`.

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Pull request merged**